integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1391,7 +1391,7 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
         return realm.create<IPerson>(FlexiblePersonSchema.name, { _id: new BSON.ObjectId(), name: "Tom", age: 36 });
       });
 
-      await realm.subscriptions.waitForSynchronization();
+      await realm?.syncSession?.uploadAllLocalChanges();
```

Comment:
This makes sense. I guess we should rename the function `addPersonAndWaitForUpload`, and change the comment to `Add a Person object and wait for the change to be uploaded`

Reply:
That's a good change 👍🏼